components/membership/membership-client-impl/src/main/kotlin/net/corda/membership/impl/client/MGMResourceClientImpl.kt

```
@@ -573,17 +588,38 @@ class MGMResourceClientImpl @Activate constructor(
             return mgm to memberShortHash
         }
 
-        private fun publishMemberInfo(memberInfo: PersistentMemberInfo, mgm: String, member: String) {
+        private fun publishMemberInfoAndRequestDistribution(
+            memberInfo: PersistentMemberInfo,
+            groupParameters: InternalGroupParameters?,
+            memberX500Name: MemberX500Name,
+            memberShortHash: String,
+            mgmHoldingIdentity: HoldingIdentity,
+            mgmShortHash: String,
+        ) {
+            val serialNumber = memberInfoFactory.create(memberInfo).serial
+            val publisher = coordinator.getManagedResource<Publisher>(PUBLISHER_RESOURCE_NAME)
+            val recordForGroupParameters = groupParameters?.let {
```

This seems incorrect. Is this publishing the group parameters for another member? Other members should receive the updated group parameters through the data distribution protocol and the MGM shouldn't attempt to write the members view of the group parameters to kafka (especially since they probably won't be on the same cluster)

---

Good spot. This should publish the MGM's view of the group parameters not the members.

---

You might be able to use `GroupParametersWriterService` for that if it reduces duplication

---

I fixed this. I didn't use the `GroupParametersWriterService` as so we can ignore errors if writing to Kafka failed.